server/src/main/java/org/apache/uniffle/server/ShuffleServer.java

```diff
@@ -259,6 +269,69 @@ private void blockUntilShutdown() throws InterruptedException {
     server.blockUntilShutdown();
   }
 
+  public ServerStatus getServerStatus() {
+    return serverStatus;
+  }
+
+  public void decommission() {
```

Comment:
why not using `public synchronize void decommission() {}`

Reply:
Not every request requires lock, such as repeated requests.

Reply:
But this is not in a critical path, the cost of `synchronize` is acceptable

Reply:
It's OK for me.